### ISSUE TITLE
Specs2 BeforeAfterAll with specs2-doobie

### ISSUE THREAD

**Issue by aeons** [tpolecat/doobie#594]
I have some stuff that needs to run before/after all checks in a specs2 specification using `doobie-specs2-0.5.0-M6` (and `specs2-4.0.0-RC4`).

However, it seems that both the before and after triggers are run before the  doobie `check`s.

I have made an example repo at https://github.com/aeons/doobie-specs2-beforeafter

Running test in that repo gives the following output:

```
sbt:doobie-specs2-beforeafter> test
beforeAll
afterAll
[info] DoobieBeforeAfter
[info] + 
[info]   Query0[(Int, Boolean)] defined at DoobieBeforeAfter.scala:17
[info]     select 42, true
[info]   + SQL Compiles and Typechecks
[info]   + C01 ?column? INTEGER (int4) NULL?  →  Int
[info]   + C02 bool     BIT     (bool) NULL?  →  Boolean
[info] Total for specification DoobieBeforeAfter
[info] Finished in 974 ms
[info] 4 examples, 6 expectations, 0 failure, 0 error
[info] Passed: Total 4, Failed 0, Errors 0, Passed 4
[success] Total time: 3 s, completed Sep 6, 2017 9:51:36 AM
```

The real issue I have is that I want to start a docker container with the database in it, and close it down afterwards. However, the docker container is shut down before the tests have finished running.

This might just be me not knowing how specs2 actually works.

**Comment by tpolecat** [comment]
Thanks for the report. We'll have a look.

**Comment by aeons** [comment]
This is fixed by #606 in 0.5.0-M13. :tada:
